beacon-chain/sync/rpc_beacon_blocks_by_range.go

```diff
@@ -126,6 +140,35 @@ func validateRangeRequest(r *pb.BeaconBlocksByRangeRequest, current primitives.S
 	return rp, nil
 }
 
+func (s *Service) validateRangeAvailability(ctx context.Context, rp rangeParams) (bool, error) {
+	startBlock := rp.start
+	bs, err := s.cfg.beaconDB.BackfillStatus(ctx)
+	if err != nil {
+		// If this is a genesis or legacy node, we retrieve the checkpoint
+		// sync block.
+		if !errors.Is(err, db.ErrNotFound) {
+			return false, err
+		}
+		cpr, err := s.cfg.beaconDB.OriginCheckpointBlockRoot(ctx)
+		if errors.Is(err, db.ErrNotFoundOriginBlockRoot) {
```

Comment:
Could you please use backfill service `AvailableBlocker` interface instead? You can initialize it exactly like stategen: https://github.com/prysmaticlabs/prysm/blob/develop/beacon-chain/node/node.go#L550